CodePush.ios.js

```diff
@@ -126,72 +126,106 @@ function checkForUpdate() {
  * releases, and displaying a standard confirmation UI to the end-user
  * when an update is available.
  */
-function sync(options = {}) {  
+function sync(options = {}, onSyncStatusChange, onDownloadProgress) {  
   var syncOptions = {
-    descriptionPrefix: " Description: ",
-    appendReleaseDescription: false,
     
     ignoreFailedUpdates: true,
-    
-    mandatoryContinueButtonLabel: "Continue",
-    mandatoryUpdateMessage: "An update is available that must be installed.",
-    
-    optionalIgnoreButtonLabel: "Ignore",
-    optionalInstallButtonLabel: "Install",
-    optionalUpdateMessage: "An update is available. Would you like to install it?",
-    
+    installMode: CodePush.InstallMode.ON_NEXT_RESTART,
     rollbackTimeout: 0,
-    
-    updateTitle: "Update available",
+    updateDialog: null,
     
     ...options 
   };
-
+  
+  onSyncStatusChange = typeof onSyncStatusChange == "function"
+    ? onSyncStatusChange
+    : function(syncStatus) {
+        switch(syncStatus) {
+          case CodePush.SyncStatus.CHECKING_FOR_UPDATE:
+            console.log("Checking for update.");
```

Comment:
Could we add a "[CodePush]" prefix to these log messages so that users could easily filter them in or out in their dev console?